contracts/src/Gateway.sol

```diff
@@ -493,199 +391,182 @@ contract Gateway is IGateway, IInitializable, IUpgradable {
     // Reference: Ethereum Yellow Paper
-    function _transactionBaseGas() internal pure returns (uint256) {
+    function v1_transactionBaseGas() internal pure returns (uint256) {
         return 21_000 + 14_698 + (msg.data.length * 16);
     }
 
-    // Verify that a message commitment is considered finalized by our BEEFY light client.
-    function _verifyCommitment(bytes32 commitment, Verification.Proof calldata proof)
-        internal
-        view
-        virtual
-        returns (bool)
-    {
-        return Verification.verifyCommitment(BEEFY_CLIENT, BRIDGE_HUB_PARA_ID_ENCODED, commitment, proof);
-    }
-
-    // Convert foreign currency to native currency (ROC/KSM/DOT -> ETH)
-    function _convertToNative(UD60x18 exchangeRate, UD60x18 multiplier, UD60x18 amount)
-        internal
-        view
-        returns (uint256)
-    {
-        UD60x18 ethDecimals = convert(1e18);
-        UD60x18 foreignDecimals = convert(10).pow(convert(uint256(FOREIGN_TOKEN_DECIMALS)));
-        UD60x18 nativeAmount = multiplier.mul(amount).mul(exchangeRate).div(foreignDecimals).mul(ethDecimals);
-        return convert(nativeAmount);
+    /// @dev Define the dust threshold as the minimum cost to transfer ether between accounts
+    function v1_dustThreshold() internal view returns (uint256) {
+        return 21_000 * tx.gasprice;
     }
 
-    // Calculate the fee for accepting an outbound message
-    function _calculateFee(Costs memory costs) internal view returns (uint256) {
-        PricingStorage.Layout storage pricing = PricingStorage.layout();
-        UD60x18 amount = convert(pricing.deliveryCost + costs.foreign);
-        return costs.native + _convertToNative(pricing.exchangeRate, pricing.multiplier, amount);
-    }
+    /*
+    *     _____   __________ .___         ________
+    *    /  _  \  \______   \|   | ___  __\_____  \
+    *   /  /_\  \  |     ___/|   | \  \/ / /  ____/§
+    *  /    |    \ |    |    |   |  \   / /       \
+    *  \____|__  / |____|    |___|   \_/  \_______ \
+    *          \/                                 \/
+    */
 
-    // Submit an outbound message to Polkadot, after taking fees
-    function _submitOutbound(Ticket memory ticket) internal {
-        ChannelID channelID = ticket.dest.into();
-        Channel storage channel = _ensureChannel(channelID);
+    uint256 public constant DISPATCH_OVERHEAD_GAS_V2 = 32_000;
 
-        // Ensure outbound messaging is allowed
-        _ensureOutboundMessagingEnabled(channel);
+    /// @dev Submit a message from Polkadot for verification and dispatch
+    /// @param message A message produced by the OutboundQueue pallet on BridgeHub
+    /// @param leafProof A message proof used to verify that the message is in the merkle tree committed by the OutboundQueue pallet
+    /// @param headerProof A proof that the commitment is included in parachain header that was finalized by BEEFY.
+    /// @param rewardAddress Account on BH to credit delivery rewards
+    function v2_submit(
```

Comment:
should it also be in IGateway.sol? Meanwhile we also need another function `isRelayed(nonce)` to check if the nonce has been relayed.

Reply:
Good catch!